### DISCUSSION TITLE
Copilot slows down vim insert mode

### DISCUSSION THREAD

**Discussion by YPCrumble** [community/community#57084] [Copilot Conversations]
### Select Topic Area

Bug

### Body

With the copilot.vim plugin installed, insert mode on vim is slowed down to a crawl. There's about a half second lag after each keypress.

Output of `:Copilot version`:

```
copilot.vim 1.8.4
Vim 9.0.0400
copilot/dist/agent.js 1.84.0
Node.js 18.15.0
```

**Comment by karimdevelops** [comment]


**Reply by YPCrumble** [reply]
@karimdevelops that looks to be only with VSCode - I'm using plain Vim and experiencing the lag.

**Comment by tpope** [comment]
If this is still happening, can you share the output of `:Copilot log`?

**Comment by YPCrumble** [comment]
Closing as this is now fixed since the last time I updated copilot. I'm not sure what release fixed it but here's my current (working) version for reference in the future in case it recurs:

```
copilot.vim 1.10.1
Vim 9.0.0400
dist/agent.js 1.105.350
Node.js 18.15.0
```

**Comment by adamtaylor13** [comment]
Sorry to necro this thread, but I had a similar issue, and mine turned out to be due to my Node version being too old. I wanted to put this here in case someone else found this. I was using Node 14 (don't ask)—and so `:Copilot version` returned an error. After profiling, the plugin seemed to keep calling something which caused a lot of slowness:

```
FUNCTION  copilot#agent#New()
    Defined: ~/.vim/plugged/copilot.vim/autoload/copilot/agent.vim:498
Called 9 times
Total time:   4.262316000
 Self time:   0.000416000

count     total (s)      self (s)
    9                 0.000018000   let opts = a:0 ? a:1 : {}
    9                 0.000158000   let instance = {'requests': {}, 'editorConfiguration': get(opts, 'editorConfiguration', {}), 'Close': function('s:AgentClose'), 'Notify': function('s:AgentNotify'), 'Request': function('s:AgentRequest'), 'Call': function('s:AgentCall'), 'Cancel': function('s:AgentCancel'), 'StartupError': function('s:AgentStartupError'), }
    9                 0.000082000   let instance.methods = extend({ 'LogMessage': function('s:LogMessage'), 'window/logMessage': function('s:LogMessage'), }, get(opts, 'methods', {}))
    9   4.261963000   0.000063000   let [command, node_version, command_error] = s:Command()
    9                 0.000018000   if len(command_error)
    9                 0.000015000     if empty(command)
    9                 0.000018000       let instance.id = -1
    9                 0.000014000       let instance.startup_error = command_error
    9                 0.000008000       return instance
```

I fixed this by running `nvm install 20 && nvm use 20` 🎉
